Public/Src/Utilities/UnitTests/TestUtilities.XUnit/JournalUtils.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using BuildXL.Native.IO;
+using BuildXL.Storage;
+using BuildXL.Storage.ChangeJournalService;
+using BuildXL.Utilities;
+using BuildXL.Utilities.Instrumentation.Common;
+using BuildXL.Utilities.VmCommandProxy;
+
+namespace Test.BuildXL.TestUtilities.Xunit
+{
+    /// <summary>
+    /// Journal utilities for unit tests.
+    /// </summary>
+    public static class JournalUtils
+    {
+        /// <summary>
+        /// Gets an instance of <see cref="IChangeJournalAccessor"/> for tests.
+        /// </summary>
+        /// <param name="volumeMap">Volume map.</param>
+        /// <returns>An instance of <see cref="IChangeJournalAccessor"/>.</returns>
+        public static Possible<IChangeJournalAccessor> TryGetJournalAccessorForTest(VolumeMap volumeMap)
+        {
+            string path = null;
+            bool usingTempFile = false;
+
+            if (!HasRelocatedTempInVm)

```

Comment:
>HasRelocatedTempInVm [](start = 17, length = 20)

Does this need to be conditional? Why not just always get a JournalAccessor based on the temp directory? On the test VM it will be on a machine local drive and it will be under the object directory in a non-VM run

Reply:
I was trying to preserve the old behavior before switching over.

---
In reply to: [294012874](https://github.com/microsoft/BuildXL/pull/457#discussion_r294012874) [](ancestors = 294012874)